hail/src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -919,24 +919,37 @@ class EmitMethodBuilder[C](
     }
   }
 
+  private def _invoke[T](_args: Param*): Code[T] = {
+    EmitCodeBuilder.scopedCode(this) { cb =>
+      val args = _args.map {
+        case cp: CodeParam =>
+          cp
+        case EmitParam(EmitCode(setup, m, pv)) =>
+          cb += setup
+          if (pv.pt.required) {
+            cb += Code.toUnit(m) // side effects???, is this even possible
+            EmitParam(EmitCode.present(pv))
+          } else {
+            val mv = cb.newField[Boolean](s"emitparam_m_${pv.pt._asIdent}", m)
+            EmitParam(EmitCode(Code._empty, mv, PCode(pv.pt, mv.mux(defaultValue(pv.pt), pv.code))))
```

Comment:
Few issues with this: I'm against this mux/default value pattern, and it is PCode-as-Code specific.

Reply:
Understood. I think I have a solution.